Source/WebCore/svg/SVGElement.cpp

```diff
@@ -284,15 +285,19 @@ Vector<WeakPtr<SVGResourceElementClient>> SVGElement::referencingCSSClients() co
     return copyToVector(m_svgRareData->referencingCSSClients());
 }
 
-void SVGElement::addReferencingCSSClient(SVGResourceElementClient& client)
+void SVGElement::addReferencingCSSClient(SVGResourceElementClient& client, const RenderElement& renderer)
 {
+    if (CheckedPtr container = dynamicDowncast<RenderSVGResourceContainer>(this->renderer()))
+        container->addReferencingCSSClient(renderer);
     ensureSVGRareData().addReferencingCSSClient(client);
 }
 
-void SVGElement::removeReferencingCSSClient(SVGResourceElementClient& client)
+void SVGElement::removeReferencingCSSClient(SVGResourceElementClient& client, const RenderElement& renderer)
 {
     if (!m_svgRareData)
         return;
+    if (CheckedPtr container = dynamicDowncast<RenderSVGResourceContainer>(this->renderer()))
+        container->removeReferencingCSSClient(renderer);
```

Comment:
Is it correct to omit this call if m_svgRareDate is null? This looks unrelated.

Reply:
If there is no rare data, there are no clients that can be removed.